### PR TITLE
Add an option to open files automatically & expand all packages when starting an assessment

### PR DESCRIPTION
**Pull request by Feuermagier**
### Checklist
- [x] I tested *all* changes and their related features locally or with a test instance of Artemis.
- [x] I documented the Java code using JavaDoc style.
- [x] I documented the changes in the Documentation (/docs).

### Motivation and Context
When starting a new assessment, one (or at least I) typically start with expanding all packages in the package explorer and finding and opening the main class. With this change, all packages get expanded automatically after starting an assessment. In addition, I've added a new option to the advanced Artemis preferences to configure whether to open no class, the main class, or all types after starting an assessment.

### Description
I had to add dependencies to a number of Eclipse packages, namely JDT packages (to find the main class) and navigator packages (to access the package explorer). By default, only the main class will be opened when starting a new assessment. If there are multiple main classes, only one is opened. While the opening of files is configurable, I don't think anyone doesn't want to have all packages expanded automatically, so I haven't added an option to disable this.

### Steps for Testing
Just start a new assessment. If it works, the main class should be opened automatically (as long as there is one) and all packages (including the default package) should be expanded.

### CODE REVIEW

**Comment by Feuermagier** [comment]
Btw, how can I run spotless locally? Running mvn spotless:check or spotless:apply in the root directory of the repository yields no format violations.

**Comment by dfuchss** [comment]
`mvn spotless:apply` should work

**Comment by dfuchss** [comment]
I'll have a look at this PR asap :)

**Comment by dfuchss** [comment]
> Btw, how can I run spotless locally? Running mvn spotless:check or spotless:apply in the root directory of the repository yields no format violations.

Strange bug. I'll have a look into that

**Comment by dfuchss** [comment]
I guess the problem is that spotless defines "<ratchetFrom>origin/main</ratchetFrom>" .. and you are not working on a feature branch :)

**Comment by dfuchss** [comment]
Thx a lot :)
I'll look at it today or tomorrow

**Comment by dfuchss** [comment]
@Feuermagier I like that feature. I'd suggest to slightly modify the Expand All option: If a main method exist, in the end the open tab should be the class containing the main method. Does that make sense ?

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/AssessmentUtilities.java [hunk 1]
```diff
@@ -334,4 +339,30 @@ public static String formatCustomPenaltyMessage(IMistakeType mistake, String cus
...
+	 * @param element the element to open
+	 * @param page    the page as part of which the editor will be opened
+	 */
+	public static void openJavaElement(IJavaElement element, IWorkbenchPage page) {
+		var path = element.getPath();
+		Display.getDefault().asyncExec(() -> {
+			try {
+				IDE.openEditor(page, ResourcesPlugin.getWorkspace().getRoot().getFile(path));
+			} catch (Exception e) {
+				Platform.getLog(AssessmentUtilities.class).error("Failed to open the java code element", e);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/AssessmentUtilities.java:354]: Please add the log as constant called `log` to the class

File: bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/AssessmentUtilities.java [hunk 2]
```diff
@@ -334,4 +339,30 @@ public static String formatCustomPenaltyMessage(IMistakeType mistake, String cus
...
+	}
+
+	/**
+	 * Retrieves the global project explorer of Eclipse
+	 *
+	 * @param page the page to search in
+	 * @return the global project explorer
+	 */
+	public static ProjectExplorer getProjectExplorer(IWorkbenchPage page) {
+		return (ProjectExplorer) page.findView("org.eclipse.ui.navigator.ProjectExplorer");
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.view/src/edu/kit/kastel/eclipse/common/view/utilities/AssessmentUtilities.java:366]: What does happen if it is not open :)

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java [hunk 3]
```diff
@@ -462,6 +469,40 @@ private void resetCombos() {
...
+				var mainType = JDTUtilities.findMainClass(projects[0]);
+				if (mainType.isPresent()) {
+					AssessmentUtilities.openJavaElement(mainType.get(), page);
+				} else {
+					Platform.getLog(ArtemisGradingView.this.getClass()).warn("No main class found");
+				}
+			}
+			}
+		} catch (JavaModelException e) {
+			Platform.getLog(ArtemisGradingView.this.getClass()).error("JDT failure", e);
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java:502]: Same here :)

**Review by dfuchss — CHANGES_REQUESTED** [review]
Only one minor change regarding the code

**Comment by Feuermagier** [comment]
> @Feuermagier I like that feature. I'd suggest to slightly modify the Expand All option: If a main method exist, in the end the open tab should be the class containing the main method. Does that make sense ?

Yeah, it should do so now. I've also made it so that only the main class is highlighted in the project explorer when opening all types.

**Comment by dfuchss** [comment]
One last point :)
Could you update the images of the settings in the documentation that they include the new options ?

**Comment by Feuermagier** [comment]
Done :)

**Review by dfuchss — APPROVED** [review]

**Pull request merged**